docs/guide/docs/main/overview.md

```diff
@@ -1,74 +1,109 @@
 
 Although the Dialog Manager could theoretically be implemented as a bunch of “If” and “Else” statements, this technique does not scale well in practice, because the unpredictability of natural dialogs increases the complexity of this kind of state machine exponentially.
 
-Botpress solves this problem by combining an extensible Visual Flow Editor with a powerful Dialog Manager, which abstracts and eliminate the real complexity behind such State Machines.
+Botpress solves this problem by combining an extensible Visual Flow Editor with a powerful Decision Engine Manager, which abstracts and eliminate the real complexity behind such State Machines.
+
+#### Visual Flow Editor
+
+This interface allows you to pre-define what the bot says to a user and how the chatbot processes and reacts to the user's response.
+
+#### Decision Engine
+
+This powerful component decides whether to follow the next step as dictated by the flow editor or give an alternate response, for example, a FAQ from the QnA module or a warning for profane language.
 
 > **Upcoming**: The Botpress core team is currently working on adding statistical decisions to the Dialog Manager, which will allow you to create even more natural experiences while also simplifying development.
 
 ## Content
 
-Now that the DM has decided that your bot should reply with a certain message, there are a couple of pieces of information required before the message can be sent out to the user: what kind of message should it send and how is that message going to be rendered on the different chat platforms?
+Now that the Dialog Manager has decided that your bot should reply with a certain message, there are a couple of pieces of information required before the message can be sent out to the user
+
+The bot needs to know the kind of message it should send and how that message is going to be rendered on the different chat platforms.
 
-That is what Content Elements are for. A Content Element can be seen as a structured object that holds information about what to say, and a Content Renderer is a function that transforms that structured object into a platform-specific message.
+That is what Content Elements are for. A Content Element can be seen as a structured object that holds information about what to say (for instance, the text phrase that will be displayed to the user), and a Content Renderer is a function that transforms that structured object into a platform-specific message (for example making some of the response text bold or italicized).
 
 The purpose of a Content Renderer is to allow developers to specify how the same information should behave (i.e., be displayed) on the different chat platforms.
 
 ## Extension Points
 
-There are two ways you are going to inject code in Botpress: **actions** and **hooks**.
+Botpress allows for customization by injecting your self-written code. The two main ways to customize Botpress in this way are by using **actions** and **hooks**.
```

Comment:
There should be a link to the custom code section

Reply:
Added